### PR TITLE
add lazy-load option

### PR DESCRIPTION
**Pull request by crashbell**
Regarding your feedback, I made some changes here. I also disabled collapse folder option as discussed. Please take a look and let me know if you see any problem with my code. Thanks!

### CODE REVIEW

File: src/view.tree.js [hunk 1]
```diff
@@ -119,17 +134,43 @@ TreeView.prototype.show = function(repo, treeData) {
 
 TreeView.prototype.syncSelection = function() {
   var tree = this.$view.find('.octotree_view_body').jstree(true)
-    , path = location.pathname
+    , path = decodeURIComponent(location.pathname)
+    , recursiveLoad = this.store.get(STORE.RECURSIVE)
```
Comment (crashbell) [inline src/view.tree.js]: duplicated `recursiveLoad = this.store.get(STORE.RECURSIVE)` with function `show` but both of them will be called from `octotree.js` and the store value `RECURSIVE` can be changed after saving options in setting so I decided to make this duplication.

Comment (buunguyen) [inline src/view.tree.js]: Sure, this is fine.

File: src/view.tree.js [hunk 2]
```diff
@@ -119,17 +134,43 @@ TreeView.prototype.show = function(repo, treeData) {
...
+      var path = (lastPath + "/" + curPath) 
+      arrResult.push(path)
+      return path
+    })
+    return arrResult
   }
+
+  var paths = recursiveLoad ? [currentPath] : createPaths(currentPath)
+
+  function openPathAtIndex (paths, index) {
```
Comment (buunguyen) [inline src/view.tree.js]: :+1: this version looks better.

**Comment by crashbell** [comment]
I closed this PR to create a new one regarding your feedback

**Pull request closed without merging**